packages/fxa-settings/src/lib/hooks/useFeatureEligibilityCheck/index.tsx

```diff
@@ -0,0 +1,24 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { sessionToken } from '../../cache';
+import { useAuthClient } from '../../../models';
+
+export async function useFeatureEligibilityByLocation(
+  feature: string
+): Promise<{ eligible: boolean }> {
+  const authClient = useAuthClient();
+  const featureEligibility = new Map<string, boolean>();
+
+  if (featureEligibility.has(feature)) {
```

Comment:
I don't know enough about hooks to know if this works.  It looks like it wouldn't since that map was just created above.  But maybe React does some magic that I need to learn about.

Reply:
No magic, just mistake 😰 Thank you for catching - I've revised to also add a loading status